File: SPCode/CS/UI/Form/PaymodeForm.cs
using System.Text.RegularExpressions;
using System.Diagnostics;
using System;
using System.Drawing.Drawing2D;
using System.Windows.Forms;
using System.Collections;
using System.Drawing;
using Microsoft.VisualBasic;
using System.Data.SqlClient;
using System.Data;
using System.Collections.Generic;
using WeifenLuo.WinFormsUI;
using Microsoft.Win32;
using WeifenLuo;
//using SoftLogik.Win.SPDataProxy;
using System.ComponentModel;


namespace SoftLogik.Win
{
	namespace UI
	{
		public partial class PaymodeForm
		{
			public PaymodeForm()
			{
				InitializeComponent();
			}
			
			protected virtual DataRowView NewPayMode()
			{
				return SPPayMode.GetEmptyPayMode().DefaultView[0];
			}
			
			protected override void OnRecordChanged(SPFormRecordUpdateEventArgs e)
			{
				base.OnRecordChanged(e);
				SPPayModeRow recordRow = (SPPayModeRow) e.DataRow;
				if (recordRow != null)
				{
					switch (e.DataState)
					{
						case @SPFormDataStates.New:
							SPPayMode.InsertPayMode(recordRow.Name, recordRow.Category, recordRow.BankID, recordRow.Note);
							break;
						case SPFormDataStates.Edited:
							SPPayMode.UpdatePayMode(recordRow.PayModeID, recordRow.Name, recordRow.Category, recordRow.BankID, recordRow.Note);
							break;
						case SPFormDataStates.Deleted:
							SPPayMode.DeletePayMode(recordRow.PayModeID);
							break;
					}
					
				}
			}
			
			protected override void OnLoad(System.EventArgs e)
			{
				
				try
				{
					ddlCategory.DataSource = SPPayModeCategories.DefaulList;
					ddlCategory.DisplayMember = "Name";
					ddlCategory.ValueMember = "Category";
				}
				catch (Exception)
				{
				}
				
				base.OnLoad(e);
			}
			protected override void OnRecordBinding(SPFormRecordBindingEventArgs e)
			{
				base.OnRecordBinding(e);
				
				e.BindingSettings.DisplayMember = "Name";
				e.BindingSettings.ValueMember = "PayModeID";
				try
				{
					e.DataSource = SPPayMode.GetPayModes();
					e.BindingSettings.NewRecordProc = new System.EventHandler(NewPayMode);
				}
				catch (Exception)
				{
				}
				
			}
			
		}
	}
	
}

File: SPCode/UI/Form/PaymodeForm.vb
Imports SoftLogik.Win.SPDataProxy
Imports System.ComponentModel

Namespace UI
    Public Class PaymodeForm

        Protected Overridable Function NewPayMode() As DataRowView
            Return SPPayMode.GetEmptyPayMode().DefaultView.Item(0)
        End Function

        Protected Overrides Sub OnRecordChanged(ByVal e As SPFormRecordUpdateEventArgs)
            MyBase.OnRecordChanged(e)
            Dim recordRow As SPPayModeRow = CType(e.DataRow, SPPayModeRow)
            If recordRow IsNot Nothing Then
                Select Case e.DataState
                    Case SPFormDataStates.[New]
                        SPPayMode.InsertPayMode(recordRow.Name, recordRow.Category, recordRow.BankID, recordRow.Note)
                    Case SPFormDataStates.Edited
                        SPPayMode.UpdatePayMode(recordRow.PayModeID, recordRow.Name, recordRow.Category, recordRow.BankID, recordRow.Note)
                    Case SPFormDataStates.Deleted
                        SPPayMode.DeletePayMode(recordRow.PayModeID)
                End Select

            End If
        End Sub

        Protected Overrides Sub OnLoad(ByVal e As System.EventArgs)

            Try
                ddlCategory.DataSource = SPPayModeCategories.DefaulList
                ddlCategory.DisplayMember = "Name"
                ddlCategory.ValueMember = "Category"
            Catch ex As Exception
            End Try

            MyBase.OnLoad(e)
        End Sub
        Protected Overrides Sub OnRecordBinding(ByVal e As SPFormRecordBindingEventArgs)
            MyBase.OnRecordBinding(e)

            With e.BindingSettings
                .DisplayMember = "Name"
                .ValueMember = "PayModeID"
            End With
            Try
                e.DataSource = SPPayMode.GetPayModes()
                e.BindingSettings.NewRecordProc = AddressOf NewPayMode
            Catch ex As Exception
            End Try

        End Sub

    End Class
End Namespace
